### PR TITLE
Use make_package for better test isolation

### PR DESCRIPTION
**Pull request by richafrank**
As per request of @atugushev in https://github.com/jazzband/pip-tools/pull/1159#pullrequestreview-427025355

**Changelog-friendly one-liner**: Use make_package for better test isolation

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: tests/test_cli_compile.py [hunk 1]
```diff
@@ -1310,26 +1310,44 @@ def test_prefer_binary_dist_even_there_is_source_dists(
...
-    fake_package_dir = path_to_url(fake_package_dir)
-    with open("requirements.in", "w") as req_in:
-        req_in.write(fake_package_dir + "\n")
-        req_in.write(fake_package_dir + "#egg=small_fake_with_deps\n")
+    packages = {
+        "test_package_1": make_package("test_package_1", version="0.1"),
+        "test_package_2": make_package(
+            "test_package_2", version="0.1", install_requires=["test-package-1"]
+        ),
+    }
```
Comment (atugushev) [inline tests/test_cli_compile.py:1328]:
```suggestion
    test_package_1 = make_package("test_package_1", version="0.1")
    test_package_2 = make_package("test_package_2", version="0.1", install_requires=["test-package-1"])
```

File: tests/test_cli_compile.py [hunk 2]
```diff
@@ -1310,26 +1310,44 @@ def test_prefer_binary_dist_even_there_is_source_dists(
...
+
+    dists_dir = tmpdir / "dists"
+
+    for pkg_name, pkg in packages.items():
```
Comment (atugushev) [inline tests/test_cli_compile.py:1332]:
```suggestion
    for pkg in (test_package_1, test_package_2):
```

File: tests/test_cli_compile.py [hunk 3]
```diff
@@ -1310,26 +1310,44 @@ def test_prefer_binary_dist_even_there_is_source_dists(
...
+        make_sdist(pkg, dists_dir)
+
+    with open("requirements.in", "w") as reqs_in:
+        reqs_in.write(
+            "file:{source_path}\n".format(source_path=packages["test_package_2"])
```
Comment (atugushev) [inline tests/test_cli_compile.py:1337]:
```suggestion
            "file:{source_path}\n".format(source_path=test_package_2)
```

File: tests/test_cli_compile.py [hunk 4]
```diff
@@ -1310,26 +1310,44 @@ def test_prefer_binary_dist_even_there_is_source_dists(
...
+        )
+        reqs_in.write(
+            "file:{source_path}#egg=test-package-2\n".format(
+                source_path=packages["test_package_2"]
+            )
```
Comment (atugushev) [inline tests/test_cli_compile.py:1342]:
```suggestion
            "file:{source_path}#egg=test-package-2\n".format(
                source_path=test_package_2
            )
```

File: tests/test_cli_compile.py [hunk 5]
```diff
@@ -1310,26 +1310,44 @@ def test_prefer_binary_dist_even_there_is_source_dists(
...
         with open("requirements.txt", "w") as reqs_out:
             reqs_out.write(output_content)
 
-    out = runner.invoke(cli, [])
+    out = runner.invoke(cli, ["--find-links", str(dists_dir)])
 
     assert out.exit_code == 0, out
-    assert fake_package_dir in out.stderr
-    assert "small-fake-a==0.1" in out.stderr
+    assert str(tmpdir / "packages" / "test_package_2") in out.stderr
```
Comment (atugushev) [inline tests/test_cli_compile.py:1352]:
```suggestion
    assert str(test_package_2) in out.stderr
```

**Comment by richafrank** [comment]
Thanks @atugushev . Your suggestions looked good to me. I pulled them in and ran black over it.

**Pull request merged**